src/controllers/AuthenticationController.ts

```diff
@@ -0,0 +1,345 @@
+import {NextFunction, Request, Response} from 'express';
+import userFriendlyMessage from '../consts/userFriendlyMessages';
+import UserService from '../services/UserService';
+import EmailService from '../services/EmailService';
+import JWTUtils from '../utils/jwtUtils';
+import PasswordUtils from '../utils/passwordUtils';
+import {
+  Payload,
+  Role,
+  UserAttributes,
+  UserCreationAttributes,
+} from '../models/User';
+
+export default class AuthenticationController {
+  private userService: UserService;
+  private emailService: EmailService;
+
+  constructor(userService: UserService, emailService: EmailService) {
+    this.userService = userService;
+    this.emailService = emailService;
+  }
+
+  async signUp(req: Request, res: Response, next: NextFunction) {
+    try {
+      const {email, password, passwordConfirmation} = req.body;
+      const user = await this.userService.getOneUserByEmail(email);
+
+      if (user) {
+        res.status(400);
+        res.json({message: userFriendlyMessage.failure.emailExists});
+        return;
+      }
+      if (password !== passwordConfirmation) {
+        res.status(400);
+        res.json({
+          message: userFriendlyMessage.failure.passwordConfirmationMismatch,
+        });
+        return;
+      }
+
+      const createdUser = await this.userService.createOneUser({
+        ...req.body,
+        role: 'Student',
+        isConfirmed: false,
+        isTemporary: false,
+      });
+      const payload: Payload = {
+        id: createdUser.id,
+        role: createdUser.role,
+        schoolId: createdUser.schoolId,
+        isConfirmed: createdUser.isConfirmed,
+        isTemporary: createdUser.isTemporary,
+      };
+      const accessToken = JWTUtils.generateAccessToken(payload, undefined, {});
+      // TODO: insert frontend url for confirm email.
+      const url = `test.com?token=${accessToken}`;
+      await this.emailService.sendConfirmEmail(email, url);
+      res.json({
+        message: userFriendlyMessage.success.createUser,
+      });
+    } catch (e) {
+      res.status(400);
+      res.json({message: userFriendlyMessage.failure.createUser});
+      next(e);
+    }
+  }
+
+  async confirmEmail(req: Request, res: Response, next: NextFunction) {
+    try {
+      const token = req.query.token as string;
+      let decoded;
+      try {
+        decoded = JWTUtils.verifyAccessToken(token) as Payload;
```

Comment:
don't typecast here typecast in the method `verifyAccessToken`, also you can just run `JWTUtils.verifyAccessToken(token)` if it doesn't verify the error will be caught. 

Then later below use the `JWTUtils.decode()` to get the payload. Separate the logic, don't use verify to get the payload.